core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -569,6 +569,24 @@ public SyncedFlushResult syncFlush(String syncId, CommitId expectedCommitId) thr
         }
     }
 
+    final boolean tryRenewSyncCommit() {
+        try (ReleasableLock lock = writeLock.acquire()) {
+            ensureOpen();
+            String syncId = lastCommittedSegmentInfos.getUserData().get(SYNC_COMMIT_ID);
+            if (syncId != null && translog.totalOperations() == 0 && indexWriter.hasUncommittedChanges()) {
+                logger.trace("start renewing sync commit [{}]", syncId);
+                commitIndexWriter(indexWriter, translog, syncId);
+                logger.debug("successfully sync committed. sync id [{}].", syncId);
+                lastCommittedSegmentInfos = store.readLastCommittedSegmentsInfo();
```

Comment:
I wonder if we should somehow do a refresh here (like we do in the un-sync'd `flush` case, to clear the version map), so that the old segments held open by the previous NRT reader are freed up in cases where `refresh_interval` was disabled or set to something biggish?